### PR TITLE
feat: DAH-2880 Sign In page timeout banner

### PR DESCRIPTION
**Pull request by cliu02**
## Description

Implement the React Sign in page banner for when a user reaches the timeout limit. 

## Jira ticket

https://sfgovdt.jira.com/browse/DAH-2880

### Version Control

- [x] branch name begins with `angular` if it contains updates to Angular code
- [x] branch name contains the Jira ticket number
- [x] PR name follows `type: TICKET-NUMBER Description` format, e.g. `feat: DAH-123 New Feature`

### Code quality

- [x] [the set of changes is small](https://google.github.io/eng-practices/review/developer/small-cls.html#what-is-small)
- [ ] all automated code checks pass (linting, tests, coverage, etc.)
- [x] code irrelevant to the ticket is not modified e.g. changing indentation due to automated formatting
- [x] if the code changes the UI, it matches the UI design exactly
- [ ] if the changes include human translations, follow the [human translations process](https://sfgovdt.jira.com/l/cp/XS1KpvE4)

### Review instructions

- [x] instructions specify which environment(s) it applies to
- [x] instructions work for PA testers
- [x] instructions have already been performed at least once

### Request review

- [x] PR has `needs review` label
- [x] Use `Housing Eng` group to automatically assign reviewers, and/or assign specific engineers
- [x] If time sensitive, notify engineers in Slack

### CODE REVIEW

File: app/assets/json/translations/react/en.json [hunk 1]
```diff
@@ -1578,7 +1578,14 @@
   "signIn.signInRequired": "Please sign in to your account to access this page.",
   "signIn.signedOutSuccessfully": "You have successfully logged out of your account.",
   "signIn.userTokenValidationTimeout": "We care about your security. We logged you out due to a connection issue. Please sign in to continue.",
+  "signIn.newAccount.title" : "Check your email to finish creating your account",
+  "signIn.newAccount.p1": "We sent a link to %{email}.",
+  "signIn.newAccount.p2": "Follow the link in your email to finish creating your account. It expires after 24 hours.",
+  "signIn.newAccount.sendEmailAgainButton": "Send email again",
+  "signIn.newAccount.sendEmailAgainButton.confirmation": "Email sent. Check your email.",
+  "signIn.newAccount.sendEmailAgainButton.error": "Something went wrong. We could not send an email. Try sending it again or refreshing the page.",
```
Comment (chadbrokaw) [inline app/assets/json/translations/react/en.json:1586]: These seem to be duplicated?

File: app/javascript/pages/sign-in.tsx [hunk 2]
```diff
@@ -17,4 +17,4 @@ const SignIn = (_props: SignInProps) => {
   )
 }
 
-export default withAppSetup(SignIn, true)
+export default withAppSetup(SignIn)
```
Comment (chadbrokaw) [inline app/javascript/pages/sign-in.tsx:20]: Why was this removed?

File: app/javascript/__tests__/pages/SignIn.test.tsx [hunk 3]
```diff
@@ -176,4 +178,23 @@ describe("<SignIn />", () => {
...
+      },
+    })
+    window.sessionStorage.setItem("alert_message_secondary", t("signOut.alertMessage.timeout"))
+    mockGetItem.mockImplementationOnce(() => t("signOut.alertMessage.timeout"))
+    render(<SiteAlert type="secondary" />)
+    expect(mockGetItem).toHaveBeenCalledWith("alert_message_secondary")
+    window.sessionStorage.setItem("newAccount", "test@test.com")
+    await renderAndLoadAsync(<SignIn assetPaths={{}} />)
+    expect(screen.getByText(t("signOut.alertMessage.timeout"))).not.toBeNull()
+  })
```
Comment (chadbrokaw) [inline app/javascript/__tests__/pages/SignIn.test.tsx:199]: Looks great!

File: app/javascript/authentication/SignInForm.tsx [hunk 4]
```diff
@@ -94,7 +94,11 @@ const SignInFormCard = ({
           {requestError}
         </AlertBox>
       )}
-      <SiteAlert type="success" />
+      {sessionStorage.getItem("alert_message_success") ? (
+        <SiteAlert type="success" />
+      ) : (
+        <SiteAlert type="secondary" />
```
Comment (chadbrokaw) [inline app/javascript/authentication/SignInForm.tsx:100]: Will there ever be a case where there is nothing in sessionStorage under `alert_message_secondary`

**Review by chadbrokaw — APPROVED** [review]
Some non-blocking questions but otherwise looks good!

**Review by cade-exygy — APPROVED** [review]
Looks good! Only thing I noticed is that the banner width spans full screen. If that is expected, then good to go!

Once this is merged, I can finish the PR for sign in redirect, using the SiteAlert pattern!
